src/core/Cline.ts

```diff
@@ -626,22 +628,61 @@ export class Cline {
 		if (responseImages && responseImages.length > 0) {
 			newUserContent.push(...formatResponse.imageBlocks(responseImages))
 		}
-
+		const wasInteractiveBrowser = (lastRelevantMessageIndex > 0) ? modifiedClineMessages[lastRelevantMessageIndex - 1].text?.includes("interactive mode") : false
+		let hadBrowserPort = '';
+		if ( wasInteractiveBrowser ) {
+			const match = modifiedClineMessages[lastRelevantMessageIndex - 1].text?.match(/\(browserPort\s*=\s*(\d+)\)/)
```

Comment:
Parsing the string doesn't feel that with the grain of how the rest of this code works. Could we make it a checkbox or an option on the settings page instead?

Reply:
I tried having Cline find a better solution and spent like $30 and its solutions never worked, this was the only way I got this to work

Reply:
Did you try the settings page?

Reply:
I'm not sure I follow

Reply:
Can we think of this as an option on the settings screen where you choose to either use the built in browser or your own that’s accessible on a specific port? And then we just have condition logic in the implementation of the browser tool based on your settings.

Reply:
I agree with the suggestion to add this feature to the settings page.  A text input to accept the port your browser is listening seems ideal. Also probably need so error handling in case it can't connect (and/or maybe a fallback to the built-in browser).

Reply:
I've updated the PR to use settings and also to launch a new browser if a remote debugging browser is not already running 👍 
![image](https://github.com/user-attachments/assets/fcd5ce83-1fe2-4501-9f72-7729d850eebd)